src/InlineInput/InlineInput.tsx

```diff
@@ -67,25 +67,28 @@ const InlineInput = React.forwardRef<HTMLInputElement, Props>(
             isValid = () => true,
             onChange,
             onChangeComplete = () => {},
-            onKeyboardIncrementAction = () => {},
-            onKeyboardDecrementAction = () => {},
+            onKeyboardIncrementAction = () => externalValue,
+            onKeyboardDecrementAction = () => externalValue,
             className = '',
         },
         ref
     ) => {
         const [internalValue, setInternalValue] = useState(externalValue);
+        const [beforeChangeValue, setBeforeChangeValue] =
+            useState(externalValue);
+        const [initialValue, setInitialValue] = useState(externalValue);
         useSynchronisationIfChangedFromOutside(externalValue, setInternalValue);
-        const onChangeIfValid = (
-            event: React.ChangeEvent<HTMLInputElement>
-        ) => {
+        const onChangeIfValid = (newValue: string) => {
             if (disabled) {
                 return;
             }
 
-            const newValue = event.target.value;
             setInternalValue(newValue);
             if (isValid(newValue)) {
-                onChange(newValue);
+                if (beforeChangeValue !== newValue) {
+                    setBeforeChangeValue(newValue);
+                    onChange(newValue);
+                }
             }
```

Comment:
I'm not able to immediately see the reason for having `beforeChangeValue`. Can you not just call `isValid(newValue)` before `setInternalValue`?